pool/chainstate_test.go

```diff
@@ -231,7 +231,10 @@ func testChainState(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 	go func() {
-		cs.handleChainUpdates(ctx)
+		err := cs.handleChainUpdates(ctx)
+		if err != nil {
+			t.Logf("unexpected error from chain update handler: %v", err)
```

Comment:
Why not `t.Fatalf` here?

Reply:
You can't invoke `t.Fatalf` from goroutines other than the main test goroutine.

Reply:
Hm. So an error returned by `handleChainUpdates` will cause the test to time out (which results in a panic). We don't specify a timeout so the default of 10 minutes will be used. I guess that's fine because it will still get caught by `run_tests.sh` and thus CI, but its kind of annoying if you hit it during development. Perhaps an `os.Exit(1)`?  Its a bit heavy handed, but I dont think it matters given that this is test code and we already know the test is failing.

Reply:
I changed it over to send the error to a buffered channel that the test checks at the end and fatals if it's non-nil.  In practice, if it exits early the test will failure due to other reasons, so it isn't strictly necessary, but it's safer to check the error.

Reply:
I updated the tests to respect the context when sending and receiving from the channel so that it will allow fatal test errors immediately upon context cancel.  This approach is more robust against other types of failures as well.

As an aside, tests should almost never be blindly sending to and receiving from internal channels without also selecting across the context.

Reply:
Nice!